reconnecting-websocket.ts

```diff
@@ -376,6 +376,7 @@ export default class ReconnectingWebSocket {
             .then(url => {
                 // close could be called before creating the ws
                 if (this._closeCalled) {
+                    this._connectLock = false;
```

Comment:
[They said](https://github.com/pladaria/reconnecting-websocket/issues/165#issuecomment-1115570830) to add `this._connectLock = false` in close also

Reply:
I think they're incorrect, though. Adding it in the `close()` method adds a subtle bug. I've added a test case `rapidly toggling connection` to capture it. You should see if you do as they suggest, the test fails.

The issue is that if you set `_connectLock = false` in the `close()` method, you open up this case:

 1. Connect. We pass the [`_connectLock` check](https://github.com/reedsy/reconnecting-websocket/blob/4f198190668cc255d95aa70ae2ee8a91410c7f47/reconnecting-websocket.ts#L351), so we start a [`_wait()`](https://github.com/reedsy/reconnecting-websocket/blob/4f198190668cc255d95aa70ae2ee8a91410c7f47/reconnecting-websocket.ts#L374)
 2. Now call `close()` which in this case sets `_connectLock = false`
 3. Call `reconnect()`, which internally [calls `_connect()`](https://github.com/reedsy/reconnecting-websocket/blob/4f198190668cc255d95aa70ae2ee8a91410c7f47/reconnecting-websocket.ts#L241). Now, since we've reset `_connectLock = false` in `close()`, we pass the `_connect()` `_connectLock` check *again*, and set up a *second* `_wait()`, while the first hasn't finished
 4. If we wait for these promises to resolve, we now create two WebSocket connections instead of one